x-pack/legacy/plugins/ml/public/datavisualizer/file_based/components/import_view/importer/importer.js

```diff
@@ -152,3 +159,32 @@ function updatePipelineTimezone(ingestPipeline) {
     }
   }
 }
+
+function createDocumentChunks(docArray) {
+  let chunks = [];
+  // chop docArray into 5000 doc chunks
+  const tempChunks = chunk(docArray, CHUNK_SIZE);
+
+  // loop over tempChunks and check that the total character length
+  // for each chunk is within the MAX_CHUNK_CHAR_COUNT.
+  // if the length is too long, split the chunk into smaller chunks
+  // based on how much larger it is than MAX_CHUNK_CHAR_COUNT
+  // note, each document is a different size, so dividing by charCountOfDocs
+  // only produces an average chunk size that should be smaller than the max length
+  for (let i = 0; i < tempChunks.length; i++) {
+    const docs = tempChunks[i];
+    const numberOfDocs = docs.length;
+
+    const charCountOfDocs = JSON.stringify(docs).length;
+    if (charCountOfDocs > MAX_CHUNK_CHAR_COUNT) {
+      // calculate new chunk size which should produce a chunk
+      // who's length is on average around MAX_CHUNK_CHAR_COUNT
+      const adjustedChunkSize = Math.floor((MAX_CHUNK_CHAR_COUNT / charCountOfDocs) * numberOfDocs);
+      const smallerChunks = chunk(docs, adjustedChunkSize);
+      chunks.push(...smallerChunks);
+    } else {
+      chunks = tempChunks;
```

Comment:
I'm not sure I follow the code correctly and understand this line. Should this be `chunks.push(tempChunks)` or am I missing something how `chunks` are put together?

Reply:
yep, this is wrong. great catch

Reply:
again, great catch. fixed in e87416b9649aae2ed0a02bedfde65943e0efd32d